doc/cascadia/Keybindings-Arguments.md

```diff
@@ -0,0 +1,291 @@
+
+Once the `IKeyBindingArgs` is built for the keybinding, we'll set it in
+`AppKeyBindings` with a updated `AppKeyBindings::SetKeyBinding` call.
+`SetKeyBinding`'s signature will be updated to take a `ActionAndArgs` instead.
+
+### Executing KeyBinding Actions with Arguments
+
+When we're handling a keybinding in `AppKeyBindings::_DoAction`, we'll trigger
+the event handlers with the `IKeyBindingArgs` we've stored in the map with the
+`ShortcutAction`.
+
+Then, in `App`, we'll handle each of these events. We set up lambdas as event
+handlers for each event in `App::_HookupKeyBindings`. In each of those
+functions, We'll inspect the `IKeyBindingArgs` parameter, and use args from its
+implementation to call callbacks in the `App` class.
+
+Instead of:
+```c++
+    void App::_HookupKeyBindings(TerminalApp::AppKeyBindings bindings) noexcept
+    {
+        // ...
+        bindings.NewTabWithProfile([this](const auto index) { _OpenNewTab({ index }); });
+    }
+```
+
+The code will look like:
+
+```c++
+    void App::_HookupKeyBindings(TerminalApp::AppKeyBindings bindings) noexcept
+    {
+        // ...
+        bindings.NewTabWithProfile([this](const auto& sender, const auto& args) { _OpenNewTab({ args.ProfileIndex() }); });
+    }
+```
+
+We could even go so far as to update `App` to have methods defined with the
+actual keybinding function signatures, so instead we'd just have:
+
+
+```c++
+    void App::_HookupKeyBindings(TerminalApp::AppKeyBindings bindings) noexcept
+    {
+        // ...
+        bindings.NewTabWithProfile({ this, &App::_OpenNewTab });
+    }
+
+    void App::_OpenNewTab(const TerminalApp::AppKeyBindings& sender, const NewTabEventArgs& args)
+    {
+        auto profileIndex = args.ProfileIndex();
+        // ...
+    }
+```
+
+### Serializing KeyBinding Arguments
+
+Similar to how we parse arguments from the json, we'll need to update the
+`AppKeyBindingsSerialization` code to be able to serialize the arguments from a
+particular `IKeyBindingArgs`.
+
+## UI/UX Design
+
+### Keybindings in the New Tab Dropdown
+
+Small modifications will need to be made to the code responsible for the new tab
+dropdown. The new tab dropdown currently also displays the keybindings for each
+profile in the new tab dropdown. It does this by querying for the keybinding
+associated with each action. As we'll be removing the old `ShortcutAction`s that
+this dropdown uses, we'll need a new way to find which key chord corresponds to
+opening a given profile.
+
+We'll need to be able to not only lookup a keybinding by `ShortcutAction`, but
+also by a `ShortcutAction` and `IKeyBindingArgs`. We'll need to update the
+`AppKeyBindings::GetKeyBinding` method to also accept a `IKeyBindingArgs`. We'll
+also probably want each `IKeyBindingArgs` implementation to define `operator==`,
+so that we can easily check if two different `IKeyBindingArgs` are the same in
+this method.
+
+## Capabilities
+### Accessibility
+
+N/A
+
+### Security
+
+N/A

```

Comment:
Isn't there some security consideration given that you're parsing things? Shouldn't this probably be fuzzed out?